sbin/woeusb

```diff
@@ -492,6 +494,23 @@ process_commandline_parameters(){
                     <<< "${parameters[0]}"
                 )"
                 ;;
+            --uefi-ntfs-img \
+            |-u)
+                shift_array parameters
+                uefi_ntfs_img="${parameters[0]}"
+
+                if \
+                    [[ \
+                        ( "${parameters[*]}" =~ '--tgt-fs NTFS' \
+                        || ( -z "$target_filesystem_type_input" && "$target_filesystem_type_input" = 'NTFS') ) \
```

Comment:
> Please elaborate on what you're trying to accomplish here

Basically, when one sets both `--tgt-fs` to `NTFS` and `-u` to path + UEFI image, I wanted to test if the argument supplied to `-u` is an existent file.

Actually, it checks the following:

- target filesystem:
   - `$target_filesystem_type_input` might be already defined (when `--tgt-fs` comes before `-u`), therefore we need to test that variable:
      1. if it is defined:
         - we need to use `-z`, because otherwise when the variable is not defined, Bash errors that the variable is unbound;
      2. and at the same time if it is set to `NTFS`.
   - When `$target_filesystem_type_input` is not set, `--tgt-fs` or `--target-filesystem` might be defined later on the command-line, therefore we need to check if `${parameters[*]}` contains them followed by ` NTFS`.

- `$uefi_ntfs_img`:
   - If `$uefi_ntfs_img` is set to a value which is non-existent file, WoeUSB should warn the user about this and revert to downloading the image from the Internet.

> runtime parameter verification should be done in the `check_runtime_parameters` function

Well, how exactly would you like me to do it? Currently, that function accepts the following three arguments:

```bash
check_runtime_parameters \
install_mode \
source_media \
target_media \
"${new_file_system_label}"
```

AFAIK, any of those parameters contain neither filesystem type (e.g. `NTFS`) nor the `$uefi_ntfs_img` variable values (obviously, as it is proposed to add to WoeUSB in this PR). Without adding those information into `check_runtime_parameters` function, I don’t see a way to implement it that way.

> This check doesn't match the `--target-filesystem` long-variant option

Oops, I missed that one.

> Please close all parameters with curly braces

Okay.

> The quoting of `'NTFS'` is unnecessary, please drop them

Okay.

Reply:
We might also want to test the supplied file using `file`:

```bash
file /git/others/rufus/res/uefi/uefi-ntfs.img 
/git/others/rufus/res/uefi/uefi-ntfs.img: DOS/MBR boot sector, code offset 0x3c+2, OEM-ID "mkfs.fat", sectors/cluster 4, root entries 512, sectors 1024 (volumes <=32 MB), Media descriptor 0xf8, sectors/FAT 1, sectors/track 32, heads 64, serial number 0xa8da13d, label: "UEFI_NTFS  ", FAT (12 bit)
```

For example, by grepping `DOS/MBR boot sector` and `label: "UEFI_NTFS`.

Reply:
> Without adding those information into check_runtime_parameters function, I don’t see a way to implement it that way.

Pass it into that function then?  We can always pass new parameters when necessary.

> We might also want to test the supplied file using `file`

I prefer not doing so, at least not in this PR.